## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 433974`

Logs: https://github.com/yzhou216/nixpkgs-review-gha/actions/runs/17013570161

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-433974-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://yzhou216.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  yzhou216.cachix.org-1:O7nGwQnLGGBGE8zYFMipVzPVN956FcV57y9SqwEP+O8=
  ' \
  /nix/store/nsprw87hrclvcw30pfk1nj8f49bm3snb-dmsdos-unstable-2021-02-06
```
</details></li>
</ul></details>


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>dmsdos</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>dmsdos</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>dmsdos</summary>
<pre>[  9%] Building C object CMakeFiles/dmsdos.dir/src/lib_interface.c.o
[ 13%] Building C object CMakeFiles/dmsdos.dir/src/dblspace_dec.c.o
/nix/build/nix-build-dmsdos-unstable-2021-02-06.drv-3549-2490609613/source/src/lib_interface.c:31:10: fatal error: 'malloc.h' file not found
   31 | #include <malloc.h>
      |          ^~~~~~~~~~
/nix/build/nix-build-dmsdos-unstable-2021-02-06.drv-3549-2490609613/source/src/dblspace_interface.c:34:10: fatal error: 'malloc.h' file not found
   34 | # include<malloc.h>
      |          ^~~~~~~~~~
/nix/build/nix-build-dmsdos-unstable-2021-02-06.drv-3549-2490609613/source/src/dblspace_dec.c:33:9: fatal error: 'malloc.h' file not found
   33 | #include<malloc.h>
      |         ^~~~~~~~~~
1 error generated.
1 error generated.
make[2]: *** [CMakeFiles/dmsdos.dir/build.make:91: CMakeFiles/dmsdos.dir/src/dblspace_interface.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
1 error generated.
make[2]: *** [CMakeFiles/dmsdos.dir/build.make:104: CMakeFiles/dmsdos.dir/src/dblspace_dec.c.o] Error 1
make[2]: *** [CMakeFiles/dmsdos.dir/build.make:78: CMakeFiles/dmsdos.dir/src/lib_interface.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:102: CMakeFiles/dmsdos.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>dmsdos</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>dmsdos</summary>
<pre>[ 13%] Building C object CMakeFiles/dmsdos.dir/src/lib_interface.c.o
[ 13%] Building C object CMakeFiles/dmsdos.dir/src/dblspace_interface.c.o
/nix/build/nix-build-dmsdos-unstable-2021-02-06.drv-3689-1232850209/source/src/lib_interface.c:31:10: fatal error: 'malloc.h' file not found
   31 | #include <malloc.h>
      |          ^~~~~~~~~~
/nix/build/nix-build-dmsdos-unstable-2021-02-06.drv-3689-1232850209/source/src/dblspace_dec.c:33:9: fatal error: 'malloc.h' file not found
   33 | #include<malloc.h>
      |         ^~~~~~~~~~
/nix/build/nix-build-dmsdos-unstable-2021-02-06.drv-3689-1232850209/source/src/dblspace_interface.c:34:10: fatal error: 'malloc.h' file not found
   34 | # include<malloc.h>
      |          ^~~~~~~~~~
1 error generated.
1 error generated.
make[2]: *** [CMakeFiles/dmsdos.dir/build.make:104: CMakeFiles/dmsdos.dir/src/dblspace_dec.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/dmsdos.dir/build.make:91: CMakeFiles/dmsdos.dir/src/dblspace_interface.c.o] Error 1
1 error generated.
make[2]: *** [CMakeFiles/dmsdos.dir/build.make:78: CMakeFiles/dmsdos.dir/src/lib_interface.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:102: CMakeFiles/dmsdos.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/dm/dmsdos/package.nix
+++ b/pkgs/by-name/dm/dmsdos/package.nix
@@ -1,0 +1,36 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dmsdos";
+  version = "unstable-2021-02-06";
+
+  src = fetchFromGitHub {
+    owner = "sandsmark";
+    repo = "dmsdos";
+    rev = "c9044d509969d3d1467b781c08233e15c1da7a13";
+    hash = "sha256-oGVkDf1gFaSMRpvHq4GNLN8htBm/sYawZeVwiqQxjL8=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  meta = {
+    description = "Linux utilities to handle dos/win95 doublespace/drivespace/stacker";
+    homepage = "https://github.com/sandsmark/dmsdos.git";
+    changelog = "https://github.com/sandsmark/dmsdos/blob/${finalAttrs.src.rev}/NEWS";
+    license = with lib.licenses; [
+      lgpl2
+      gpl2Plus
+    ];
+    maintainers = with lib.maintainers; [ matthewcroughan ];
+    mainProgram = "dmsdos";
+    platforms = lib.platforms.all;
+    badPlatforms = lib.platforms.darwin;
+  };
+})

